contributor-docs/adrs/adr-XXX-file-structure.md

```diff
@@ -0,0 +1,42 @@
+# ADR XXX: File structure
+
+## Status
+
+Proposed
+
+## Context
+
+Components might be more grokable if they were structured consistently. This ADR proposes conventions
+
+## Decision
+
+TL;DR:
+
+```
+primer-react/
+├─ src/
+│  ├─ BreadCrumbs/
+│  │  ├─ index.ts                    // Just re-exporting?
+│  │  ├─ BreadCrumbs.tsx             // Primary component
+│  │  ├─ BreadCrumbsItem.tsx         // Subcomponent (include item name for findability in most IDEs)
+│  │  ├─ BreadCrumbs.mdx             // Documentation. Always .mdx, not .md
+│  │  ├─ BreadCrumbs.stories.tsx
+│  │  ├─ BreadCrumbs.test.tsx        // Unit tests
+│  │  ├─ BreadCrumbs.types.test.tsx  // Type tests
+│  │  ├─ BreadCrumbs.yml             // Component metadata (Possible future)
+│  │  └─ __snapshots__/
+┆  ┆
+```
+
+### Rules
+
+- Every component should have its own PascalCased directory directly under `src/`
+- Subcomponents meant to be used as children of their parent component should be properties of the exported component (e.g., `BreadCrumbs.Item`)
+- Subcomponents meant to be used on their own should be exported as a named export (e.g., `ButtonDanger`)
```

Comment:
Is this a pattern we want to encourage?

I'd argue that it's not a "subcomponent" if it's exported on its own. If it's not a property of a parent component, I think it should probably have it's own directory.

Reply:
I like simple rules.... but thinking about that with `ButtonDanger`.... that doesn't feel like it merits its own component, and I'm not sure I'd want that to just be a prop on Button (i.e., `<Button variant="danger"...`). Would you say `ButtonDanger` should have its own directory?

Reply:
>  I'm not sure I'd want that to just be a prop on Button (i.e., <Button variant="danger"...)

`ButtonDanger` is going away in favor of a `variant` prop. @pksjce's `Button2` uses that API.

Reply:
Okay cool. That settles it then :-)